bcl2fastq/index_summary.py

```diff
@@ -93,6 +93,7 @@ def update_known_index_bar(run_json):
     run['library'] = run['SampleID'].str.extract(
         'SWID_\d+_(\w+_\d+_.*_\d+_[A-Z]{2})_'
     )
+    run['Index1'] = run['Index1'].astype('str')
```

Comment:
As I read it, this casting is done to convert `NaN` into the string `nan` and to allow the concatenation to occur in the next line.

I am not sure this is the best approach, as `nan` can be seen as a valid (if degenerate) DNA sequence. Replacing the `NaN` with an empty string would be the clearer approach.

Reply:
To confirm -- the filtering should happen right after the assignment to `index` happens?

Reply:
Correct. 

This is a global variable shared by all connections to Dashi, so it makes sense to do sanity checks at this stage and avoid recalculations for each connection.